### PR TITLE
Streamline exception catching in Http Responses

### PR DESCRIPTION
**Pull request by alexispeter**
Idea: Streamline the configuration and code of catching exceptions in Responses' `process()`.

Current config:
```php
    $config->CM_Http_Response_Page->catch = array(
        'CM_Exception_Nonexistent'  => ['path' => '/error/not-found', 'log' => true],
        'CM_Exception_InvalidParam' => ['path' => '/error/not-found', 'log' => true],
        'CM_Exception_AuthRequired' => ['path' => '/error/auth-required', 'log' => false],
        'CM_Exception_NotAllowed'   => ['path' => '/error/not-allowed', 'log' => false],
    );

    $config->CM_Http_Response_View_Abstract->catch = array(
        'CM_Exception_Nonexistent',
        'CM_Exception_AuthRequired',
        'CM_Exception_NotAllowed',
        'CM_Exception_Blocked',
        'CM_Exception_ActionLimit',
    );

    $config->CM_Http_Response_RPC->catch = array(
        'CM_Exception_AuthRequired',
        'CM_Exception_NotAllowed',
    );
```

Idea:
```php
    $config->CM_Http_Response_Abstract->exceptionsToCatch = array(
        'CM_Exception_Nonexistent'  => ['errorPage' => 'CM_Page_Error_NotFound', 'log' => 'CM_Paging_Log_NotFound'],
        'CM_Exception_InvalidParam' => ['errorPage' => 'CM_Page_Error_NotFound', 'log' => 'CM_Paging_Log_NotFound'],
        'CM_Exception_AuthRequired' => ['errorPage' => 'CM_Page_Error_AuthRequired', 'log' => null],
        'CM_Exception_NotAllowed'   => ['errorPage' => 'CM_Page_Error_NotAllowed', 'log' => null],
        'CM_Exception_Blocked'      => ['errorPage' => 'CM_Page_Error_NotFound', 'log' => null],
        'CM_Exception_ActionLimit'  => ['errorPage' => 'CM_Page_Error_NotFound', 'log' => null],
    );
```

@tomaszdurka @dlondero @mariansollmann thoughts?

### CODE REVIEW

**Comment by alexispeter** [comment]
please review @njam

**Comment by njam** [comment]
@tomaszdurka could you review this?

**Comment by njam** [comment]
@tomaszdurka could you review this?

**Comment by njam** [comment]
@fauvel could you review this?

**Comment by fauvel** [comment]
Reviewing

File: library/CM/Http/Response/Page.php [hunk 1]
```diff
@@ -132,7 +132,7 @@ protected function _processPageLoop(CM_Http_Request_Abstract $request) {
      * @return string|null|boolean
      */
     private function _processPage(CM_Http_Request_Abstract $request) {
-        try {
+        return $this->_runWithCatching(function () use ($request) {
```
Comment (fauvel) [inline library/CM/Http/Response/Page.php]: Fyi `use ($request)` actually **clones** the object (!)
I would say that it's best practice to pass it by reference explicitly unless you really want to clone it.

**Comment by fauvel** [comment]
Lgtm!

**Comment by alexispeter** [comment]
tagged https://github.com/cargomedia/CM/releases/tag/1.105.0

**Pull request merged**